interoperator/pkg/controller/schedulers/sfroundrobinscheduler/sfroundrobinscheduler_controller.go

```diff
@@ -0,0 +1,139 @@
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSFRoundRobinScheduler{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	cfgManager, err := config.New(mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	interoperatorCfg := cfgManager.GetConfig()
+	if interoperatorCfg.SchedulerType != constants.RoundRobinSchedulerType {
+		return nil
+	}
+	// Create a new controller
+	c, err := controller.New("sfroundrobinscheduler-controller", mgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: interoperatorCfg.SchedulerWorkerCount,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to SFServiceInstance
+	err = c.Watch(&source.Kind{Type: &osbv1alpha1.SFServiceInstance{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileSFRoundRobinScheduler{}
+
+// ReconcileSFRoundRobinScheduler reconciles a SFServiceInstance object
+type ReconcileSFRoundRobinScheduler struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a SFRoundRobinScheduler object and makes changes based on the state read
+// and what is in the SFRoundRobinScheduler.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  The scaffolding writes
+// a Deployment as an example
+func (r *ReconcileSFRoundRobinScheduler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the SFRoundRobinScheduler instance
+	instance := &osbv1alpha1.SFServiceInstance{}
+	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	if instance.Spec.ClusterID == "" {
+		clusters := &resourcev1alpha1.SFClusterList{}
+		options := &kubernetes.ListOptions{}
+		err := r.List(context.TODO(), options, clusters)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		items := clusters.Items
+		sort.Slice(items, func(i, j int) bool {
+			if items[i].GetCreationTimestamp().Time == items[j].GetCreationTimestamp().Time {
+				return items[i].Name < items[j].Name
+			}
+			return !items[i].GetCreationTimestamp().After(items[j].GetCreationTimestamp().Time)
+		})
+		if len(items) <= lastProvisionedClusterIndex {
+			lastProvisionedClusterIndex = 0
+		}
+		currentlyProvisionedCluster := items[lastProvisionedClusterIndex]
+		lastProvisionedClusterIndex++
```

Comment:
Depending on the length of the worker queue, the `Reconcile` func could be called in parallel. So, the section using `lastProvisionedClusterIndex` might have to be synchronized.

Reply:
@subhankarc working on it

Reply:
done.